rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -54,9 +74,57 @@ public BookKeeper(Configuration conf)
     }
 
     @Override
-    public List<Boolean> getCacheStatus(String remotePath, long fileLength, long lastModified, long startBlock, long endBlock)
+    public List<com.qubole.rubix.bookkeeper.Location> getCacheStatus(String remotePath, long fileLength, long lastModified, long startBlock, long endBlock, int clusterType)
             throws TException
     {
+        if (clusterManager == null || currentNodeIndex == -1) {
```

Comment:
Move this code under a Supplier and add define a new config to tell the refresh time (keep default as 10sec). 
Also, check if we can make Supplier to reload in background instead of blocking the get() request
Finally, use same config in all cluster managers to drive their Supplier memoization time